src/spacyMatcher.py

```diff
@@ -1,54 +1,49 @@
-## Function: spacyMatcher
-## Input: TEXT and TAG (PERSON, ORG, GPE, etc)
-##        If TAG = '' will return all TAGs 
-## Output: List of dictionaries for all TAGs found in the text (os a specific tag if passed)
-## Author: The Sentimentalists / Ana B Potje
-## Date: 15/Oct/2020
-##
-## This function uses the Python Library Spacy. The user will input an URL and a "TAG" such as:
-##    PERSON - People, including fictional.
-##    ORG - Companies, agencies, institutions, etc.
-##    GPE - Countries, cities, states.
-##    PERCENT - Percentage, including ”%“.
-##    LANGUAGE - Any named language.
-##    DATE - Absolute or relative dates or periods.
-##    TIME - Times smaller than a day.
-##    LOC - Non-GPE locations, mountain ranges, bodies of water.
-##    NORP - Nationalities or religious or political groups.
-##    EVENT - Named hurricanes, battles, wars, sports events, etc.
-##    WORK_OF_ART - Titles of books, songs, etc.
-##    MONEY - Monetary values, including unit.
-##    QUANTITY - Measurements, as of weight or distance.
-##    ORDINAL - “first”, “second”, etc.
-##    CARDINAL - Numerals that do not fall under another type (not ordinal, quantity ..)
-##
-## It will return a list with distinct values of all occurrences of that entity in the text.
-###########################################################################################################
-
+"""
+This module contains a a function that returns named entities from text
+"""
 from aws_xray_sdk.core import xray_recorder
+import spacy
 
 @xray_recorder.capture('spacyMatcher')
-def spacyMatcher(text, tag): 
-    import spacy
-
-    if tag == '':
-        all_tags = ['PERSON','GPE','ORG','PERCENT','LANGUAGE','DATE','TIME','LOC','NORP','EVENT','WORK_OF_ART',
+def spacyMatcher(text, tag=None):
+    """
+    This function takes text and an optional tag.
```

Comment:
I wasn't clear what the input text actually is. It looks to me like it should be the raw text you want to analyse but you said in your comment that it's a URL, that doesn't look right to me.

Reply:
Oh I see ... In the original code the URL was the input. 
Then the function code was refactored, and it gets a raw text as the input. 
I simply didn't update the comment - apologies ...